Sofa/Component/Constraint/Lagrangian/Correction/src/sofa/component/constraint/lagrangian/correction/LinearSolverConstraintCorrection.inl

```diff
@@ -596,15 +617,14 @@ void LinearSolverConstraintCorrection<DataTypes>::addConstraintDisplacement(doub
 
             for (MatrixDerivColConstIterator colIt = rowIt.begin(); colIt != rowEnd; ++colIt)
             {
-                const auto dof = colIt.index();
-                Deriv disp(type::NOINIT);
-
-                for(Size j = 0; j < derivDim; j++)
+                if (systemLHVector_buf_fullvector)
```

Comment:
As this is constant for all loop iterations a code design moving the if out of the two for() scope would make things more clear.